src/Fuzz.elm

```diff
@@ -107,29 +115,20 @@ custom : Generator a -> Shrinker a -> Fuzzer a
 custom generator shrinker =
     let
         shrinkTree a =
-            Rose a (Lazy.List.map shrinkTree (shrinker a))
+            Rose a (Lazy.lazy <| \_ -> Lazy.force <| Lazy.List.map shrinkTree (shrinker a))
```

Comment:
This change made a huge performance difference. The `Shrink` library returns lazy lists, but the process of building them isn't necessarily lazy. For instance, the int shrinker builds a list of shrunken ints, then turns it into a lazy list.

This change ensures we only pay the cost when we actually start shrinking, but even when shrinking we might not be interested in all the shrunken ints a call to `Shrink.int` generates, so further performance improvements may be obtained by changes in the `Shrink` lib. The easiest strategy for this I found is copying the `Shrink` library into the `elm-test` codebase temporarily, making changes, and then running benchmarks.